File: ILT.IHR.APP/ClientApp/src/app/core/interfaces/RolePermission.ts
export class IRolePermissionDisplay {
    rolePermissionID: number;
    roleID: number;
    roleName: string;
    moduleID: number;
    moduleShort: string;
    moduleName: string;
    view: boolean;
    add: boolean;
    update: boolean;
    delete: boolean;
}

export class IRolePermission {
    RolePermissionID: number;
    RoleID: number;
    ModuleID: number;
    View: boolean;
    Add: boolean;
    Update: boolean;
    Delete: boolean;
    CreatedBy: string;
    CreatedDate: Date;
    ModifiedBy: string;
    ModifiedDate: Date;
    TimeStamp: string;
}

File: ILT.IHR.DTO/RolePermission.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.ComponentModel.DataAnnotations;

namespace ILT.IHR.DTO
{
    public class RolePermission: AbstractDataObject
    {

		public int RolePermissionID
		{
			get { return base.RecordID; }
			set { base.RecordID = value; }
		}
		public int RoleID { get; set; }
		public string RoleName { get; set; }
		[Required]
		[Range(1, 100000)]
		public int ModuleID { get; set; }
		public string ModuleShort { get; set; }
		public string ModuleName { get; set; }
		public bool View { get; set; }
		public bool Add { get; set; }
		public bool Update { get; set; }
		public bool Delete { get; set; }

	}
}
